vm/systemSmartContracts/validator.go

```diff
@@ -1064,17 +1065,21 @@ func (v *validatorSC) stake(args *vmcommon.ContractCallInput) vmcommon.ReturnCod
 		}
 	}
 
-	v.activateStakingFor(
-		blsKeys,
-		registrationData,
-		validatorConfig.NodePrice,
-		registrationData.RewardAddress,
-		args.CallerAddr,
-	)
-
-	if v.isNumberOfNodesTooHigh(registrationData) {
-		v.eei.AddReturnMessage("number of nodes is too high")
-		return vmcommon.UserError
+	if !v.isNumberOfNodesTooHigh(registrationData) {
```

Comment:
if I have 100 keys registered, but only 10 staked, would it activate the next key?

Reply:
Added unit test for this scenario